### PR TITLE
doc: nRF Machine learning app: restructured the doc

### PR DESCRIPTION
**Pull request by divipillai**
Resturucted the doc by:

* Splitting the doc into subpages
* Updated the doc to match the applicaiton template
* Review of the content

### CODE REVIEW

File: applications/machine_learning/app_desc.rst [hunk 1]
```diff
@@ -0,0 +1,524 @@
...
+Data forwarding requirements
+============================
+
+To forward the collected data using `Edge Impulse's data forwarder`_, you must install the Edge Impulse CLI.
+See `Edge Impulse CLI installation guide`_ for instructions.
+
+Nordic UART Service requirements
+--------------------------------
+
+If you want to forward data over Nordic UART Service (NUS), you need an additional development kit that is able to run the :ref:`central_uart` sample.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:165]: Link to NUS (nus_service_readme)

**Review by peknis — CHANGES_REQUESTED** [review]

File: applications/machine_learning/app_desc.rst [hunk 2]
```diff
@@ -0,0 +1,524 @@
...
+
+The nRF5340 DK uses RTT as the logger's backend.
+The RTT logs can be easily accessed, because the DK has a built-in SEGGER chip.
+
+Custom model requirements
+=========================
+
+The default application configurations rely on pre-trained machine learning models that can be automatically downloaded during the application build.
+If you want to train and deploy a custom machine learning model using `Edge Impulse Studio`_, you need a user account for the Edge Impulse Studio web-based tool.
+The user account is not needed to perform predictions using the pre-trained models.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:154]:
```suggestion
You do not need a user account to perform predictions using the pre-trained models.
```

File: applications/machine_learning/app_desc.rst [hunk 3]
```diff
@@ -0,0 +1,524 @@
...
+
+Data forwarding requirements
+============================
+
+To forward the collected data using `Edge Impulse's data forwarder`_, you must install the Edge Impulse CLI.
+See `Edge Impulse CLI installation guide`_ for instructions.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:160]:
```suggestion
See the `Edge Impulse CLI installation guide`_ for instructions.
```

File: applications/machine_learning/app_desc.rst [hunk 4]
```diff
@@ -0,0 +1,524 @@
...
+**************
+
+The application supports a simple user interface.
+You can control the application using predefined buttons, while LEDs are used to display information.
+
+Buttons
+=======
+
+The application supports a button that is used to switch between data forwarding and running the machine learning model.
+You can trigger the change by pressing and holding the button for longer than 5 seconds.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:192]:
```suggestion
To trigger the change, press and hold the button for longer than five seconds.
```

File: applications/machine_learning/app_desc.rst [hunk 5]
```diff
@@ -0,0 +1,524 @@
...
+
+.. note::
+   If a given configuration supports Bluetooth, then two sequences of the button press and release in a short time interval, shortly after application boot, removes the Bluetooth bonds.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:195]:
```suggestion
   If a given configuration supports Bluetooth, two sequences of the button press and release in a short time interval, right after application boot, removes the Bluetooth bonds.
```

File: applications/machine_learning/app_desc.rst [hunk 6]
```diff
@@ -0,0 +1,524 @@
...
+* Thingy:53 displays the application state in the RGB scale using **LED1**.
+
+  * If the device is returning the machine learning prediction results, the LED uses the following predefined colors:
+
+    * ``rotate`` - Red
+    * ``updown`` - Green
+    * ``tap`` - Blue
+    * Anomaly - Purple
+
+    If the machine learning model is running but it has not detected anything yet or the ``idle`` state is detected, the LED is blinking.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:232]:
```suggestion
    If the machine learning model is running but has not detected anything yet or the ``idle`` state is detected, the LED is blinking.
```

File: applications/machine_learning/app_desc.rst [hunk 7]
```diff
@@ -0,0 +1,524 @@
...
+
+.. _nrf_machine_learning_app_requirements_build_types:
+.. _nrf_machine_learning_app_configuration_build_types:
+
+nRF Machine Learning build types
+================================
+
+The nRF Machine Learning application does not use a single :file:`prj.conf` file.
+Before you start testing the application, you can select one of the build types supported by the application.
+Not every board supports both mentioned build types.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:325]:
```suggestion
All boards do not support both mentioned build types.
```

File: applications/machine_learning/app_desc.rst [hunk 8]
```diff
@@ -0,0 +1,524 @@
...
+     - :file:`prj_rtt.conf`
+     - ``thingy53_nrf5340_cpuapp`` and ``thingy53_nrf5340_cpuapp_ns``
+     - Debug version of the application that uses RTT for printing logs instead of USB CDC.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`applications/machine_learning`
+
+The nRF machine learning application is built the same way as any other |NCS| application or sample.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:361]:
```suggestion
The nRF Machine Learning application is built the same way as any other |NCS| application or sample.
```

File: applications/machine_learning/app_desc.rst [hunk 9]
```diff
@@ -0,0 +1,524 @@
...
+Selecting a build type
+======================
+
+Before you start testing the application, you can select one of the :ref:`nrf_machine_learning_app_requirements_build_types`.
+See :ref:`modifying_build_types` for detailed steps on how to select a build type.
+
+Providing API key
+=================
+
+If the URI of the Edge Impulse zip file requires providing an additional API key, you can provide it using the following CMake definition: :c:macro:`EI_API_KEY_HEADER`.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:375]:
```suggestion
If the URI of the Edge Impulse :file:`.zip` file requires providing an additional API key, you can provide it using the :c:macro:`EI_API_KEY_HEADER` CMake definition.
```

File: applications/machine_learning/app_desc.rst [hunk 10]
```diff
@@ -0,0 +1,524 @@
...
+This definition is set in a similar way as the selected build type.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:376]:
```suggestion
This definition is set in a similar way to the selected build type.
```

File: applications/machine_learning/app_desc.rst [hunk 11]
```diff
@@ -0,0 +1,524 @@
...
+=======
+
+After programming the application to your development kit, you can test the nRF Machine Learning application.
+You can test running the machine learning model on an embedded device and forwarding data to `Edge Impulse studio`_.
+The detailed test steps for the DKs and the Thingy:53 are described in the following subsections.
+
+Application logs
+----------------
+
+In most of the provided debug configurations, the application provides logs through the RTT.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:393]:
```suggestion
In most of the provided ``debug`` configurations, the application provides logs through the RTT.
```

File: applications/machine_learning/app_desc.rst [hunk 12]
```diff
@@ -0,0 +1,524 @@
...
+#. Move the device up and down.
+   The ``updown`` gesture is recognized by the machine learning model.
+   The LED color changes to green, and the LED stays turned on.
+#. Rotate the device.
+   The ``rotate`` gesture is recognized by the machine learning model.
+   The LED color changes to red, and the LED stays turned on.
+#. Shake the device.
+   The machine learning model detects an anomaly.
+   The LED color changes to purple, and the LED stays turned on.
+#. Press and hold the button for more than 5 seconds to switch to the data forwarding mode.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:422]:
```suggestion
#. Press and hold the button for more than five seconds to switch to the data forwarding mode.
```

File: applications/machine_learning/app_desc.rst [hunk 13]
```diff
@@ -0,0 +1,524 @@
...
+   After the mode is switched, the LED color changes to red, and the LED starts blinking very slowly.
+#. Program the :ref:`central_uart` sample to a compatible development kit, for example, the nRF52840 DK.
+#. Turn on the programmed device.
+   After a brief delay, the Bluetooth® connection between the sample and the Thingy is established.
+   The Thingy forwards the sensor readouts over NUS.
+   The LED on the Thingy starts to blink rapidly.
+#. Connect to the Bluetooth® Central UART sample with a terminal emulator (for example, PuTTY).
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:429]:
```suggestion
#. Connect to the sample with a terminal emulator (for example, PuTTY).
```

File: applications/machine_learning/app_desc.rst [hunk 14]
```diff
@@ -0,0 +1,524 @@
...
+   See :ref:`putty` for the required settings.
+#. Observe the sensor readouts represented as comma-separated values.
+   Every line represents a single sensor readout.
+   The Thingy forwards sensor readouts over NUS to the Central UART sample.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:433]:
```suggestion
   The Thingy forwards sensor readouts over NUS to the sample.
```

File: applications/machine_learning/app_desc.rst [hunk 15]
```diff
@@ -0,0 +1,524 @@
...
+   Initially, **LED2** displays the LED effect representing ``sine`` wave (1 blink), and **LED1** is breathing, because the signal was not yet recognized by the machine learning model.
+   After a brief delay, the machine learning model recognizes the simulated signal.
+   **LED1** and **LED2** display the same LED effect.
+#. Press **Button 3** to change the generated acceleration signal.
+   Right after the signal change, the effects displayed by LEDs are different.
+   After a brief delay, the machine learning model recognizes the ``triangle`` wave, and the same effect (2 blinks) is displayed by both LEDs.
+#. Press **Button 3** to again change the generated acceleration signal.
+   The ``square`` wave (3 blinks) is displayed only by the **LED2**.
+   This signal is marked as an anomaly by the machine learning model, and **LED1** starts breathing.
+#. Press and hold **Button 1** for more than 5 seconds to switch to the data forwarding mode.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:456]:
```suggestion
#. Press and hold **Button 1** for more than five seconds to switch to the data forwarding mode.
```

File: applications/machine_learning/app_desc.rst [hunk 16]
```diff
@@ -0,0 +1,524 @@
...
+   After the mode is switched, **LED1** starts to blink rapidly.
+#. Connect to the development kit with a terminal emulator (for example, PuTTY).
+   See :ref:`putty` for the required settings.
+#. Observe the sensor readouts represented as comma-separated values.
+   Every line represents a single sensor readout.
+#. Turn off PuTTY to ensure that only one program will access data on UART.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:462]:
```suggestion
#. Turn off PuTTY to ensure that only one program has access to the data on UART.
```

File: applications/machine_learning/app_desc.rst [hunk 17]
```diff
@@ -0,0 +1,524 @@
...
+Optionally, you can also connect to the device using `Edge Impulse's data forwarder`_ and forward data to `Edge Impulse studio`_ (after logging in).
+See `Forwarding data to Edge Impulse studio`_ for details.
+
+Forwarding data to Edge Impulse studio
+--------------------------------------
+
+To start forwarding data to Edge Impulse studio:
+
+1. Make sure you meet the `Data forwarding requirements`_ before forwarding data to Edge Impulse studio.
+#. Run the ``edge-impulse-data-forwarder`` Edge Impulse command line tool.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:473]:
```suggestion
#. Run the ``edge-impulse-data-forwarder`` Edge Impulse command-line tool.
```

File: applications/machine_learning/app_desc.rst [hunk 18]
```diff
@@ -0,0 +1,524 @@
...
+
+         Sampling example
+
+.. _nrf_machine_learning_app_porting_guide:
+
+Porting guide
+*************
+
+You can port the nRF Machine Learning application to any board available in the |NCS| or Zephyr.
+To do so, create the board-specific directory in :file:`applications/machine_learning/configuration/` and add the application configuration files there.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:500]:
```suggestion
Create the board-specific directory in :file:`applications/machine_learning/configuration/` and add the application configuration files there.
```

File: applications/machine_learning/app_desc.rst [hunk 19]
```diff
@@ -0,0 +1,524 @@
...
+    If the machine learning model is running but has not detected anything yet or has detected an anomaly, the **LED1** is breathing.
+  * If the device forwards data, the **LED1** uses the following blinking patterns:
+
+    * LED blinks slowly if it is not connected.
+    * LED blinks with an average frequency if it is connected but is not actively forwarding data.
+    * LED blinks rapidly if it is connected and is actively forwarding data.
+
+.. _nrf_machine_learning_app_configuration:
+
+Configuration
```
Comment (grochu) [inline applications/machine_learning/app_desc.rst:262]: Just a suggestion, I don't know what templates say about it, but e.g. nRF Dekstop has a separate page covering configuration: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/nrf_desktop/application_kconfig.html#nrf-desktop-application-specific-kconfig-options
In fact Machine Learning has a few Kconfig files in src/ src/modules and src/utils so maybe there is enough content.

**Review by grochu — COMMENTED** [review]
The update looks good, just a comment about the overall architecture

File: applications/machine_learning/app_desc.rst (hunk 19 above)
Comment (divipillai) [reply applications/machine_learning/app_desc.rst:262]: Good point. I will create a new page for the configuration options. We follow the same for nRF5340 Audio applications as well - https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/applications/nrf5340_audio/index.html.

**Review by grochu — APPROVED** [review]
Looks good, thanks, for refactoring these docs.

File: applications/machine_learning/config_options.rst [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+
+nRF Machine Learning: Application-specific Kconfig options
+##########################################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The nRF Machine Learning introduces Kconfig options that you can use to simplify an application configuration.
+The application introduces application-specific configuration options related to its modules and data providers.
```
Comment (peknis) [inline applications/machine_learning/config_options.rst:11]:
```suggestion
The nRF Machine Learning introduces Kconfig options that you can use to simplify the application configuration.
The application introduces configuration options related specifically to its modules and data providers.
```

File: applications/machine_learning/config_options.rst [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+You can enable and configure the modules separately for the selected board and build type.
+See the documentation page of the selected module for information about the functionalities provided by the module and its configuration.
+See :ref:`nrf_machine_learning_app_internal_modules` for a list of modules available in the application.
+
+Advertising configuration
+*************************
+
+If a given build type enables Bluetooth, the :ref:`caf_ble_adv` is used to control the Bluetooth advertising.
+This CAF module relies on :ref:`bt_le_adv_prov_readme` to manage advertising data and scan response data.
+The nRF Machine Learning application configures the data providers in :file:`src/util/Kconfig`.
```
Comment (peknis) [inline applications/machine_learning/config_options.rst:26]:
```suggestion
The data providers used by the nRF Machine Learning application are configured using the :file:`src/util/Kconfig` file.
```

File: applications/machine_learning/config_options.rst [hunk 22]
```diff
@@ -0,0 +1,32 @@
...
+By default, the application enables a set of data providers available in the |NCS| and adds a custom provider that appends UUID128 of Nordic UART Service (NUS) to the scan response data if the NUS is enabled in the configuration and the Bluetooth local identity in use has no bond.
+
+Following are the application-specific configuration options that can be configured for the nRF Machine Learning and the modules:
```
Comment (peknis) [inline applications/machine_learning/config_options.rst:29]:
```suggestion
Following are the application-specific configuration options that you can configure for the nRF Machine Learning and the modules:
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: applications/machine_learning/app_desc.rst [hunk 23]
```diff
@@ -0,0 +1,516 @@
...
+* :file:`prj_<build_type>.conf` - Kconfig configuration file for a :ref:`custom build type <modifying_build_types>`.
+  To support a given build type for the selected board, you must define the configuration file with a proper name.
+  See :ref:`nrf_machine_learning_app_configuration_build_types` for more information.
+* :file:`app.overlay` - DTS overlay file specific for the board.
+  Defining the DTS overlay file for a given board is optional.
+* :file:`_def` files - These files are defined separately for modules used by the application.
+  You must define a :file:`_def` file for every module that requires it and enable it in the configuration for the given board.
+  The :file:`_def` files that are common for all the boards and build types are located in the :file:`applications/machine_learning/configuration/common` directory.
+
+The nRF Machine Learning application introduces application-specific Kconfig options that can be used to simplify an application configuration.
```
Comment (peknis) [inline applications/machine_learning/app_desc.rst:293]:
```suggestion
The nRF Machine Learning application introduces application-specific Kconfig options that you can use to simplify the application configuration.
```

File: applications/machine_learning/config_options.rst [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+
+The nRF Machine Learning application is modular and event driven.
+You can enable and configure the modules separately for the selected board and build type.
+See the documentation page of the selected module for information about the functionalities provided by the module and its configuration.
+See :ref:`nrf_machine_learning_app_internal_modules` for a list of modules available in the application.
+
+Advertising configuration
+*************************
+
+If a given build type enables Bluetooth, the :ref:`caf_ble_adv` is used to control the Bluetooth advertising.
```
Comment (peknis) [inline applications/machine_learning/config_options.rst:24]:
```suggestion
If the given build type enables Bluetooth, the :ref:`caf_ble_adv` is used to control the Bluetooth advertising.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by peknis — APPROVED** [review]

**Pull request merged**